triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -782,13 +820,17 @@ class Runner(
 
 object Runner extends StrictLogging {
 
-  /** The number of submitSingleCommand invocations each trigger will
-    * attempt to execute in parallel.  Note that this does not in any
-    * way bound the number of already-submitted, but not completed,
-    * commands that may be pending.
-    */
-  val maxParallelSubmissionsPerTrigger = 8
-  val maxTriesWhenOverloaded = 6
+  private final case class RunnerSubmissionConfig(maxRequests: Int, maxDuration: FiniteDuration)
+
+  private final case class RunnerConfig(
+      parallelism: Int,
+      maxRetries: Int,
+      maxInflightCommands: Int,
+      submission: RunnerSubmissionConfig,
+  )
+
+  private val config: RunnerConfig =
+    ConfigSource.resources("trigger-config.conf").at("trigger").loadOrThrow[RunnerConfig]
```

Comment:
Loads `trigger` config namespace into the `RunnerConfig` case class. This loads once when the `Runner` companion object is loaded by the JVM. Any config load failure throws and so should cause the CLI/service to crash.

Reply:
With the case class `'Int` types, we could introduce refined types to further validate (at config load time) that these values were (say) > 0.

Reply:
These changes will be moved to a separate PR

Reply:
PR for trigger runner configuration is https://github.com/digital-asset/daml/pull/15482